### DISCUSSION TITLE
Possible to use protocol buffers with airframe RPC?

### DISCUSSION THREAD

**Discussion by rogern** [wvlet/airframe#3335] [Q&A]
I'm wondering if it at all is possible without too much effort to use protocol buffers with airframe RPC? Our codebase is already well invested in protobufs and changing to MessagePack is simply not an option for us.

**Comment by xerial** [comment]
Unfortunately no. Although airframe-grpc is a relatively small code, mapping ProtocolBuffer (schema-less) data to methods and objects in Scala will need more effort. 

A potential future direction would be generating plain gRPC-java server stub from .proto file as well as a plain Airframe RPC interface/impl code in Scala to bridge gRPC and Airframe RPC servers.

ScalaPB might be an easier path for supporting Protobuf https://scalapb.github.io/docs/grpc
